### ISSUE TITLE
習題 6: 請為 macrograd 加上一個 crossEntropyLoss 層，然後用 mnist 測試

### ISSUE THREAD

**Issue by ccckmit** [ccc112b/ai#4]
修改前的參考

1. https://github.com/newcodevelop/micrograd/ (請從這版開始修改）

改完後的參考

1. [03b-MacroGrad/mnist.ipynb](https://github.com/ccc112b/py2gpt/blob/master/03b-MacroGrad/mnist.ipynb)
2. [03b-MacroGrad/mnist.py](https://github.com/ccc112b/py2gpt/blob/master/03b-MacroGrad/mnist.py)
3. [03b-MacroGrad/macrograd/engine.py](https://github.com/ccc112b/py2gpt/blob/master/03b-MacroGrad/macrograd/engine.py) （參考答案）

理論的參考

1. [py2gpt/wiki](https://github.com/cccbook/py2gpt/wiki)
2. [py2gpt/wiki/softmax](https://github.com/cccbook/py2gpt/wiki/softmax)
3. https://mattpetersen.github.io/softmax-with-cross-entropy

**Comment by LeeYi-user** [comment]
https://github.com/LeeYi-user/ai/tree/master/homework/06
資工三 李易 111010512

**Comment by KiuBios** [comment]
資工二 111110519 胡劭家
https://github.com/KiuBios/ai/tree/master/hw6

**Comment by tom71012** [comment]
資工三 吳承泰 111010505
[習題6](https://github.com/tom71012/ai/tree/master/%E7%BF%92%E9%A1%8C/Homework06)

**Comment by 111010557** [comment]
資工三 楊欣潔1110105557
https://github.com/111010557/ai/blob/master/hw/06

**Comment by lkhagvaa2444** [comment]
資工二 李荷微葦 111110544
https://github.com/lkhagvaa2444/ai/tree/master/HW6

**Comment by ctfpanda17** [comment]
電三甲 111010148 林晏霆
https://github.com/ctfpanda17/ai/tree/master/hw6

**Comment by jackuio440** [comment]
資工三 111010502 劉爵寬
https://github.com/jackuio440/ai/tree/master/HW/06

**Comment by weixiang0470** [comment]
資工三 黃偉祥 111010550
[Homework 6](https://github.com/weixiang0470/ai112b/tree/master/Homework/hw06)

**Comment by Kruez0** [comment]
資工二劉佩芬111110541
https://github.com/Kruez0/ai/tree/master/h6

**Comment by jiajianong** [comment]
資工四 110910535 呂嘉融
[習題六](https://github.com/jiajianong/ai/blob/master/%E7%BF%92%E9%A1%8C%E5%85%AD/hw6)

**Comment by RogerChen530** [comment]
資工三111010530陳樂融
[Week6](https://github.com/RogerChen530/AIHomework/tree/main/Week6)

**Comment by wrr606** [comment]
資工二 111110517 范揚玄
https://github.com/wrr606/ai/tree/master/homework/6

**Comment by MitanEXE** [comment]
資工三 111010544 陳彬彬
[作業6](https://github.com/MitanEXE/ai/tree/master/HW6)

**Comment by ali1234-56** [comment]
資工三 111010536 林家成
[hk6](https://github.com/ali1234-56/ai-homework/tree/main/hk6)

**Comment by Tzuan1020** [comment]
資工三 111010517 蘇子安
[習題六](https://github.com/Tzuan1020/ai/blob/master/hw6.py)

**Comment by Remilia517** [comment]
資工三 111010532 李昱宏
[習題6](https://github.com/Remilia517/ai/tree/master/homework/6)

**Comment by Jung217** [comment]
資工三 簡志融 111010501
[習題六](https://github.com/Jung217/ai/tree/master/%E7%BF%92%E9%A1%8C/06/mine)

**Comment by rossen1020** [comment]
111010516 資工三 賴映羽
[06](https://github.com/rossen1020/ai/tree/master/%E7%BF%92%E9%A1%8C/06)

**Comment by sam00002135** [comment]
資工三 111010574 白和鑫
https://github.com/sam00002135/ai/blob/master/home6.py

**Comment by kingpider123** [comment]
資工三 111010552 阮銘福
https://github.com/kingpider123/ai/tree/master/homework/6

**Comment by yihsien924** [comment]
資工三 林羿嫻 111010555
[hw6](https://github.com/yihsien924/ai/tree/master/hw6)

**Comment by ruoxin030704** [comment]
資工三 111010576 王若馨
https://github.com/ruoxin030704/ai/tree/master/6

**Comment by henk3520** [comment]
資工三 古莛禾 111010573
https://github.com/henk3520/ai/tree/main/hw6

**Comment by byby9527** [comment]
111010540 資工三 陳先正
https://github.com/byby9527/ai/tree/master/06

**Comment by weichen11011** [comment]
資工三 顏瑋成 111010529
[習題六](https://github.com/weichen11011/ai/blob/master/hw6/micrograd_withCrossEntropy.py)

**Comment by Chieh0622** [comment]
資工三 111010515 林弘杰
[HW6](https://github.com/Chieh0622/ai/tree/master/HW06)

**Comment by a920217** [comment]
資工三 111010538 陳紹恩
[hw6](https://github.com/a920217/ai/blob/master/hw6.py)

**Comment by mimmm4308** [comment]
資工三 111010525 林品翔
[HW6](https://github.com/mimmm4308/ai/tree/master/06)

**Comment by gakuplusq** [comment]
[110910530 資工四 黃劭騏](https://github.com/gakuplusq/ai/wiki/%E7%BF%92%E9%A1%8C%E5%85%AD)

**Comment by Ellinaa** [comment]
資工四 110910558 陳玟卉
[習題6](https://github.com/Ellinaa/ai/blob/master/EX/6.py)

**Comment by XaioYue** [comment]
資工二 111110555 董翔宇
https://github.com/XaioYue/ai/tree/master/H6

**Comment by patrick0516** [comment]
資工三 111010575 周辰恩
[hw6](https://github.com/patrick0516/ai_report/tree/master/hw6)

**Comment by yehchen0328** [comment]
資工三 111010535 葉哲辰
https://github.com/yehchen0328/ai/blob/master/hw6_mni.py
https://github.com/yehchen0328/ai/blob/master/hw6_eng.py

**Comment by guangbo1110105228** [comment]
資工三111010528 冉光博
https://github.com/guangbo1110105228/ai/blob/master/hw6_engine.py
https://github.com/guangbo1110105228/ai/blob/master/hw6_mnsit.py

**Comment by nelson023** [comment]
資工四 110910523張繼光
https://github.com/nelson023/ai/tree/master/hw/hw6

**Comment by abc644** [comment]
資工四 110910521 張展睿
https://github.com/abc644/ai/tree/master/HW/hw6
